R/glmClassical.R

```diff
@@ -0,0 +1,1199 @@
+           dof     = modelObj$df.residual,
+           pval    = pchisq(pearson,
+                            df=modelObj$df.residual,
+                            lower.tail=FALSE))
+    )
+  }
+}
+
+# GLM estimates table
+.glmEstimatesTable <- function(jaspResults, dataset, options, ready, position) {
+  if (!options$coefEstimates || !is.null(jaspResults[["estimatesTable"]]))
+    return()
+
+  estimatesTable <- createJaspTable(gettext("Coefficients"))
+  estimatesTable$dependOn(optionsFromObject   = jaspResults[["modelSummary"]],
+                          options             = c("coefEstimates", "coefCi", "coefCiInterval"))
+  estimatesTable$position <- position
+  estimatesTable$showSpecifiedColumnsOnly <- TRUE
+
+  if (options$family %in% c("bernoulli", "binomial", "poisson"))
+    testStat <- "z"
+  else
+    testStat <- "t"
+
+  estimatesTable$addColumnInfo(name = "param",    title = "", type = "string")
+  estimatesTable$addColumnInfo(name = "est",      title = gettext("Estimate"), type = "number")
+  estimatesTable$addColumnInfo(name = "se",       title = gettext("Standard Error"), type = "number")
+  estimatesTable$addColumnInfo(name = "testStat", title = gettext(testStat), type = "number")
+  estimatesTable$addColumnInfo(name = "pval",     title = gettext("p"), type = "pvalue")
+
+  if (options$coefCi) {
+    ciPercentage <- options$coefCiInterval * 100
+    if (floor(ciPercentage) == ciPercentage)
+      ciPercentage <- as.integer(ciPercentage)
+    ciTitle <- paste(ciPercentage, " % ", "Confidence Interval",sep = "")
+    estimatesTable$addColumnInfo(name = "ciLow",    title = gettext("Lower Bound"), type = "number", overtitle = ciTitle)
+    estimatesTable$addColumnInfo(name = "ciUpp",    title = gettext("Upper Bound"), type = "number", overtitle = ciTitle)
+  }
+
+  jaspResults[["estimatesTable"]] <- estimatesTable
+  .glmEstimatesTableFill(jaspResults, dataset, options, ready)
+}
+
+.glmEstimatesTableFill <- function(jaspResults, dataset, options, ready) {
+  if (!ready)
+    return()
+  # compute glm models
+  glmModels <- .glmComputeModel(jaspResults, dataset, options)
+  modelSummary <- summary(glmModels[["fullModel"]])[["coefficients"]]
+  rowNames <- rownames(modelSummary)
+
+  if (options$coefCi) {
+    coefCiSummary <- confint(glmModels[["fullModel"]], level = options$coefCiInterval)
+  } else {
+    coefCiSummary <- matrix(nrow = length(rowNames),
+                            ncol = 2,
+                            data = rep(0, length(rowNames)*2))
+  }
+
+  for (i in seq_along(rowNames)) {
+    jaspResults[["estimatesTable"]]$addRows(
+      list(param     = jaspBase::gsubInteractionSymbol(rowNames[i]),
+           est       = modelSummary[i, "Estimate"],
+           se        = modelSummary[i, "Std. Error"],
+           testStat  = modelSummary[i, 3],
+           pval      = modelSummary[i, 4],
+           ciLow     = coefCiSummary[i, 1],
+           ciUpp     = coefCiSummary[i, 2])
+    )
+  }
```

Comment:
```suggestion
  for (rowName in rowNames) {
    jaspResults[["estimatesTable"]]$addRows(
      list(param     = jaspBase::gsubInteractionSymbol(rowName),
           est       = modelSummary[rowName, "Estimate"],
           se        = modelSummary[rowName, "Std. Error"],
           testStat  = modelSummary[rowName, 3],
           pval      = modelSummary[rowName, 4],
           ciLow     = coefCiSummary[rowName, 1],
           ciUpp     = coefCiSummary[rowName, 2])
    )
  }
```

It seems like you can simplify this by just looping over the row names instead of the row indices. Also, what is the reason you don't use `$setData` here? Seems that should work just fine here!

Reply:
I will try $setData. 
I didn't subset the columns by the column names, because the column names can change depending on the distribution family used (e.g. t.stat and z.stat). I think it's safe to just use the indices here since the unit test will catch it if the column order is different.

Reply:
Ok, I see the point. Yes, the unit tests would catch it if this breaks, but the reason for using the column names would be that the tests would not fail in the first place :) But ok, let's not obsess about this here. Is the column name also different for pvalues and CIs with different families? If not, you can use the column name at least for those.

Reply:
Yes, the columns names are different for test statistics, p-values and CIs, depending on the family or the level of confidence.

Reply:
Yes, this seems fine to me. You are right that you are still assuming a particular order of the columns, but you were doing that anyways :)

Reply:
btw `jaspBase::gsubInteractionSymbol` is able to take a vector of strings, so you do not need to `sapply` over the `rowNames`